### PR TITLE
Add option to disable tools (currently exiv2 binary)

### PR DESCRIPTION
**Pull request by a17r**
In a multiarch build, one may want to build binaries only for native arch.

### CODE REVIEW

**Comment by piponazo** [comment]
I think it is a good idea to add this variable to disable the generation of the apps. I would even control with that variable the generation of the samples.

I think many developers might be only interested in the usage of the libraries and they might not be interested at all in compiling the applications.

If @clanmills agrees, I will propose you some ideas to improve this Pull Request.

**Comment by a17r** [comment]
Yes I thought about samples as well and ignored it for now - it is unfortunate that a subdirectory labelled 'unit tests' is installing stuff at all - and is seemingly also required to generate doc.

**Comment by clanmills** [comment]
With the autotools build, the default build is the library + about 10 little programs usesd by the test suite: exiv2/exifprint/addmoddel/etc..  $ make samples builds the other samples.

I believe the CMake build defaults to building the libraries + exiv2 + samples + test programs.  Perhaps well could polish this to be:

1 default builds = library + exiv2 application
2 option to omit exiv2 application
3 we bring the test programs up-to-date when we run the test suite
4 option to build everything (library + exiv2, test programs and samples)

However I don't have any strong feelings about this.  The library is about 75000 lines of C++.  The samples and test programs are small (10000 lines of C++). 

What is _**"unit tests"**_?

**Comment by a17r** [comment]
Took that from

``OPTION( EXIV2_ENABLE_BUILD_SAMPLES "Build the unit tests" ON )``

**Review by piponazo — CHANGES_REQUESTED** [review]

File: src/CMakeLists.txt [hunk 1]
```diff
@@ -285,9 +285,11 @@ set( EXIV2_SRC  exiv2.cpp
                 utils.hpp
 )
 
-add_executable( exiv2 ${EXIV2_SRC} )
-target_link_libraries( exiv2 PRIVATE exiv2lib )
-install(TARGETS exiv2 RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR})
+if(EXIV2_ENABLE_TOOLS)
```
Comment (clanmills) [inline src/CMakeLists.txt]: You have the correct idea in reorganizing the code.  We really should have src/ for the library code (test/src=test applications) and samples/src for sample applications.  However, this is lots of messing about.  Your thoughts involving  `if (EXIV2_ENABLE_TOOLS)` is the quick fix.

I'm got a couple of other things to suggest:
1) It should be called EXIV2_ENABLE_BUILD_EXIV2APP "Build exiv2 command line program"
2) The help string for EXIV2_ENABLE_BUILD_SAMPLES "Build sample applications"

**Pull request merged**